### PR TITLE
Fixes msvc2015 detection when only vs2019 are installed.

### PR DESCRIPTION
**Pull request by lygstate**
Signed-off-by: Yonggang Luo <luoyonggang@gmail.com>

Fixed:
https://github.com/microsoft/vscode-cmake-tools/issues/1905

### CODE REVIEW

**Comment by andreeis** [comment]
@lygstate, we got to review this for 1.8. I merged with the latest changes in develop and I am testing.
Even your first commit did not include util.envSet and util.envGetValue in util.ts. First I thought that I merged something wrong.
I am did some changes on my side to unblock testing but if you had anything special in mind for envSet and engGetValue please push an update.

**Comment by andreeis** [comment]
> Also fixes the output encoding.
@lygstate, the output encoding fix is in util.parseVersion? Do we have a GitHub issue created for that?

File: src/kit.ts
Comment (andreeis) [inline src/kit.ts:854]: @lygstate, did you have anything in mind with moving the whole targetArch === "arm" or "arm64" inside the if majorVersion < 14? It causes arm/arm64 kits to be created in cmake-user-kits.json even when arm/arm64 is not installed.

Comment (lygstate) [reply src/kit.ts:854]: Indeed, I don't know why this is needed, as for  majyorVersion>=14, we can always using 'vcvarsall.bat'.
And for vs2010, 'vcvarsall.bat' also exist. I even think vcvarsScript = `vcvars${kitHostTargetArch(hostArch, targetArch, true)}.bat`; should be removed totally, I don't know why this present

**Comment by lygstate** [comment]
> > Also fixes the output encoding.
> 
> @lygstate, the output encoding fix is in util.parseVersion? Do we have a GitHub issue created for that?

I split the fix of encoding problem in separate PR https://github.com/microsoft/vscode-cmake-tools/pull/1985

File: src/kit.ts
Comment (andreeis) [reply src/kit.ts:854]: The reason for invoking vcvars target custom name is that, when arm/arm64 is not installed, that script will not exist. If installed, the script will exist. If we use vcvarsall, it exists regardless of arm/arm64 installed and also it doesn't fail when run. On top of that, it produces INCLUDE and other env vars so it will cause arm/arm64 kits to be created, listed even if there is no arm/arm64 support in the VS installation. Users can be very confused by this.

**Comment by andreeis** [comment]
@lygstate, if you want you can push an update. I tested this and I sign off with 2 comments:
- revert the vcvars file name change as it was before this PR, for arm32/arm64
- figure out what you wanted with util.envGetValue and util.envSet. For now, when testing, I simply used var.get and var.set.
If you won't have the chance, I'll push an update or start a new PR from my own branch (derived from yours) later tomorrow.

**Comment by andreeis** [comment]
@lygstate, I noticed that you mentioned that this PR depends on https://github.com/microsoft/vscode-cmake-tools/pull/1763. We may not be able to test thoroughly the env var consistency PR before we release 1.8. It was a large PR with interesting scenarios to cover. When I tested this msvc PR it worked fine. Let me know if I overlooked something. Do you think this msvc PR can work without the env var consistency one?

Another question, can you describe the encoding problem? Just to make sure I properly test that one too and I don't miss something. What was happening.

**Comment by andreeis** [comment]
@lygstate, for an example of what I needed to change in order to have successful testing see branch dev/andris/cmake-tools/fromPR1955.

**Review by andreeis — APPROVED** [review]

**Comment by bobbrow** [comment]
@lygstate Is it necessary to add 'Platform' to the list of MSVC variables to make your PR succeed?  We are experiencing a regression in building some of our projects and this blocks the release of 1.8.0.  @elahehrashedi has PR #2056 out where she removes this from the list and it fixes our problem.  Do you see any problem with us removing it?

**Pull request merged**
